src/fiber.cr

```diff
@@ -85,16 +89,18 @@ class Fiber
 
   # :nodoc:
   def initialize(@stack : Void*)
+    @running = true
     @proc = Proc(Void).new { }
     @stack_top = _fiber_get_stack_top
-    @stack_bottom = GC.stack_bottom
+    @stack_bottom = GC.current_thread_stack_bottom
     @name = "main"
 
     @@fibers.push(self)
   end
 
   # :nodoc:
   def run
+    GC.enable
```

Comment:
Hmm, why is this necessary? Where is the corresponding `disable`?

Reply:
It's on the other comment you left ;-)

Reply:
No, that one reenables itself, unless there are some really strange shenanigans going on.

Reply:
Ah, you are right. I think this one might be off.

Reply:
The world is suspended during the context switch: the scheduler switched context, which means that it called `GC.disable` in the current fiber then jumped to `Fiber#run` of the new fiber, and it will only call `GC.enable` when the current fiber is eventually resumed.